packages/runtime/src/agent/file_storage.ts

```diff
@@ -18,32 +29,33 @@ export class FileStorageAgent {
     return Deno.core.opAsync(
       "op_file_storage_agent_simple_put",
       {
-        url: url instanceof URL ? url.href : url,
+        path,
         ensureDir: options?.ensureDir ?? false,
+        configurationId: this.configurationId,
       },
       byteArray
     );
   }
 
-  async delete(url: URL | string): Promise<void> {
-    return Deno.core.opAsync(
-      "op_file_storage_agent_delete",
-      url instanceof URL ? url.href : url
-    );
+  async delete(path: string): Promise<void> {
+    return Deno.core.opAsync("op_file_storage_agent_delete", {
+      path,
+      configurationId: this.configurationId,
+    });
   }
 
-  async list(url: URL | string): Promise<Array<FileStorage.FileType>> {
-    return Deno.core.opAsync(
-      "op_file_storage_agent_list",
-      url instanceof URL ? url.href : url
-    );
+  async list(path: string): Promise<Array<FileStorage.FileType>> {
+    return Deno.core.opAsync("op_file_storage_agent_list", {
+      path,
+      configurationId: this.configurationId,
+    });
   }
 
-  async createDirAll(url: URL | string): Promise<void> {
-    return Deno.core.opAsync(
-      "op_file_storage_agent_create_dir_all",
-      url instanceof URL ? url.href : url
-    );
+  async createDirAll(path: string): Promise<void> {
+    return Deno.core.opAsync("op_file_storage_agent_create_dir_all", {
+      path,
+      configurationId: this.configurationId,
```

Comment:
I think it will be better to put configurationId before path and other arguments.